hollow/src/main/java/com/netflix/hollow/api/producer/fs/HollowFilesystemBlobStager.java

```diff
@@ -110,6 +111,63 @@ public HollowFilesystemBlobStager(File stagingPath, BlobCompressor compressor) {
         return new FilesystemBlob(fromVersion, toVersion, REVERSE_DELTA, stagingPath, compressor, optionalPartConfig);
     }
 
+    @Override
+    public HollowProducer.HeaderBlob openHeader(long version) {
+        return new FilesystemHeaderBlob(version, stagingPath, compressor);
+    }
+
+    public static class FilesystemHeaderBlob extends HeaderBlob {
```

Comment:
From a very superficial view, it seems that a bunch of duplicate code could be avoided if `HeaderBlob` and `HollowProducer.Blob` implemented a common interface (say `PublishArtifact`). Is that something you've already considered? So handling header vs. snapshot et al could look something like-
```
if (publishArtifact instanceof HeaderBlob) {
    // do somethign with header
} else if (publishArtifact instanceof HollowProducer.Blob) {
    HollowProducer.Blob blob = (HollowProducer.Blob) publishArtifact;
    switch (blob.ype) {
        case SNAPSHOT: // do something with snapshot; break;
        ...
    }
}
```

Reply:
Did some refactoring following our discussion